samples/Kconfig

```diff
@@ -16,14 +16,58 @@ config NCS_SAMPLES_DEFAULTS
 	help
 	  Use the default configuration for NCS samples.
 
-config NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE
-	bool "Enable empty_app_core as a child image"
-	default n
+if SOC_SERIES_NRF53X
+
+config NCS_CHILD_IMAGE_SAMPLE_EMPTY_APP_CORE
+	bool "Add empty_app_core as a child image"
 	depends on SOC_NRF5340_CPUNET
 	select PARTITION_MANAGER_ENABLED
 	help
-	  Add the empty_app_core as a child image for the given sample.
-	  Used by samples that only require SOC_NRF5340_CPUNET.
+	  Add empty_app_core as a child image (application core).
+	  Used by samples that run on the network core only.
+
+config NCS_AUTO_INCLUDE_NET_CORE_CHILD_IMAGE
+	bool "Automatically add a child image to the network core when relevant"
+	depends on SOC_NRF5340_CPUAPP
+	select PARTITION_MANAGER_ENABLED
+	default y if BT_RPMSG_NRF53 || NRF_802154_SER_HOST || BT_RPC
+	help
+	  Add a child image to the network core, depending on the current
+	  configuration.
```

Comment:
```suggestion
This setting includes an RPMsg sample on the nRF5340 netcore.
The sample included depends on the configuration of the application.
The following RPMsg samples may be included when this setting is enabled:
In order of precedence:
- Multiprotocol RPMsg: When nRF IEEE 802.15.4 Serialization host is enabled
- HCI RPMsg: When Bluetooth is enable
- nRF IEEEE 802.15.4 Serialization RPMsg: When bluetooth is disabled
- nRF RPC Host: When Bluetooth and Bluetooth serialization over RPC is enabled
```
Disclaimer, I'm unsure I got the correct dependencies described in text so please verify those first.
This just shows that if it is difficult for us to figure out when / where then it will be even harder for our users, and therefore documenting this in the help text is very important.